### PR TITLE
Fixed SpatialMesh disapearing when AllowMeshVisualizationControl enabled

### PR DESCRIPTION
**Pull request by dannycarrera**
I noticed if you have DrawVisualMeshes enabled and AllowMeshVisualizationControl enabled, the SpatialMesh Visual would turn off due to the ToggleSpatialMesh method.

### CODE REVIEW

**Comment by keveleigh** [comment]
I'm not sure this change aligns with my expectation of how those two settings would work together.
Scenarios:
1. I want my app to always display spatial mapping.
a. DrawVisualMeshes = true
b. AllowMeshVisualizationControl = false
2. I want my app to start displaying spatial mapping, but turn off after I place.
a. DrawVisualMeshes = true
b. AllowMeshVisualizationControl = true
3. I want my app to only display spatial mapping while I'm placing.
a. DrawVisualMeshes = false
b. AllowMeshVisualizationControl = true
4. I never want spatial mapping.
a. DrawVisualMeshes = false
b. AllowMeshVisualizationControl = false

I'm not sure this change allows scenario 2 to happen anymore.

**Comment by dannycarrera** [comment]
I was considering this as well, I didn't take scenario 2 into account. I made this change originally because before, no matter what, if AllowMeshVisualzationControl = true, after placing the VisualMesh would turn off, even if DrawVisualMeshes = true.

**Comment by StephenHodgson** [comment]
I even suggest moving the logic for toggling the spatial meshes over into the `SpatialMappingManager`.

**Comment by dannycarrera** [comment]
Thanks for the feedback @keveleigh . I reviewed why I made this change in the first place. It was to fix the following issue:

Scenario 1:
    DrawVisualMeshes = true
    AllowMeshVisualizationControl = false

Expected: Always display spatial Mapping
Actual: Spatial Mapping is disabled when you begin placing.

My other assumption was that the SpatialMappingManager's settings should override TapToPlaces' setting. I see now that it is more beneficial to allow TapToPlace to override SpatialMappingManager's setting.

With that in mind, the fix is even easier than I thought.

@StephenHodgson Thanks for the suggestion! Not sure what I could move there. The logic is contained in DrawVisualMeshes to update rendering. If you had something in mind, I'd love to hear about it so I can contribute more :).

**Comment by keveleigh** [comment]
@dannycarrera Ahh! I didn't realize there was still a fix to be made; good catch! This looks good to me.

File: Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs [hunk 1]
```diff
@@ -18,8 +18,7 @@ namespace HoloToolkit.Unity.InputModule
     [RequireComponent(typeof(Interpolator))]
     public class TapToPlace : MonoBehaviour, IInputClickHandler
     {
-        [Tooltip("Distance from camera to keep the object while placing it.")]
-        public float DefaultGazeDistance = 2.0f;
+        [Tooltip("Distance from camera to keep the object while placing it.")]        public float DefaultGazeDistance = 2.0f;
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: please revert formatting

Comment (dannycarrera) [reply Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: Done

**Pull request merged**